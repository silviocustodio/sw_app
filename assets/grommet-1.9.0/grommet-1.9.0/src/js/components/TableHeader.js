// (C) Copyright 2016 Hewlett Packard Enterprise Development LP

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Button from './Button';
import Box from './Box';
import AscIcon from './icons/base/LinkDown';
import DescIcon from './icons/base/LinkUp';

export default class TableHeader extends Component {
  _onSort (index) {
    const { onSort, sortAscending, sortIndex } = this.props;
    let nextAscending;
    if (index !== sortIndex) {
      nextAscending = false;
    } else {
      nextAscending = ! sortAscending;
    }
    onSort(index, nextAscending);
  }

  render () {
    const { labels, onSort, sortAscending, sortIndex, ...props } = this.props;

    const cells = labels.map((label, index) => {
      let content;
      let options = {};

      if (Array.isArray(label)) {
        [content, options = {}] = label;
      } else {
        content = label;
        options.sortable = !!onSort;
      }

      if (sortIndex >= 0) {
        let sortIndicator;
        if (index === sortIndex) {
          sortIndicator = (
            sortAscending ?
              <AscIcon /> : <DescIcon />
          );
        }
        content = (
          <Box direction='row' justify='start' align='center'
            pad={{ between: 'small' }}>
            <span>{content}</span>
            {sortIndicator}
          </Box>
        );

        if (options.sortable) {
          content = (
            <Button plain={true} fill={true}
              onClick={this._onSort.bind(this, index)}>
              {content}
            </Button>
          );
        }
      }

      return <th key={index}>{content}</th>;
    });

    return (
      <thead {...props}>
        <tr>
          {cells}
        </tr>
      </thead>
    );
  }
}

TableHeader.propTypes = {
  labels: PropTypes.arrayOf(PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.array
  ])).isRequired,
  onSort: PropTypes.func, // (index, ascending?)
  sortAscending: PropTypes.bool,
  sortIndex: PropTypes.number
};
