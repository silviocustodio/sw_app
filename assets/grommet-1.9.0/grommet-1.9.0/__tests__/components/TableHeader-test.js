// (C) Copyright 2014-2016 Hewlett Packard Enterprise Development LP

import React from 'react';
import renderer from 'react-test-renderer';

import TableHeader from '../../src/js/components/TableHeader';

// needed because this:
// https://github.com/facebook/jest/issues/1353
jest.mock('react-dom');

describe('TableHeader', () => {
  it('has correct default options', () => {
    const component = renderer.create(
      <TableHeader labels={['first', 'second']}
        sortIndex={0} sortAscending={true} onSort={() => {}} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('accepts options for sorting', () => {
    const component = renderer.create(
      <TableHeader labels={['first', ['second', { sortable: false }]]}
        sortIndex={0} sortAscending={true} onSort={() => {}} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
