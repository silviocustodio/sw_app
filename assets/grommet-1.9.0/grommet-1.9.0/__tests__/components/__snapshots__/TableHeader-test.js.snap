// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableHeader accepts options for sorting 1`] = `
<thead>
  <tr>
    <th>
      <button
        aria-label={undefined}
        className="grommetux-button grommetux-button--fill grommetux-button--plain"
        disabled={false}
        href={undefined}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        type="button"
      >
        <div
          className="grommetux-box grommetux-box--direction-row grommetux-box--justify-start grommetux-box--align-center grommetux-box--responsive grommetux-box--pad-between-small"
          id={undefined}
          onClick={undefined}
          role={undefined}
          style={Object {}}
          tabIndex={undefined}
        >
          <span>
            first
          </span>
          <svg
            aria-label="link-down"
            className="grommetux-control-icon grommetux-control-icon-link-down grommetux-control-icon--responsive"
            height="24px"
            role="img"
            version="1.1"
            viewBox="0 0 24 24"
            width="24px"
          >
            <path
              d="M12,2 L12,22 M3,11 L12,2 L21,11"
              fill="none"
              stroke="#000"
              strokeWidth="2"
              transform="matrix(1 0 0 -1 0 24)"
            />
          </svg>
        </div>
      </button>
    </th>
    <th>
      <div
        className="grommetux-box grommetux-box--direction-row grommetux-box--justify-start grommetux-box--align-center grommetux-box--responsive grommetux-box--pad-between-small"
        id={undefined}
        onClick={undefined}
        role={undefined}
        style={Object {}}
        tabIndex={undefined}
      >
        <span>
          second
        </span>
      </div>
    </th>
  </tr>
</thead>
`;

exports[`TableHeader has correct default options 1`] = `
<thead>
  <tr>
    <th>
      <button
        aria-label={undefined}
        className="grommetux-button grommetux-button--fill grommetux-button--plain"
        disabled={false}
        href={undefined}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        type="button"
      >
        <div
          className="grommetux-box grommetux-box--direction-row grommetux-box--justify-start grommetux-box--align-center grommetux-box--responsive grommetux-box--pad-between-small"
          id={undefined}
          onClick={undefined}
          role={undefined}
          style={Object {}}
          tabIndex={undefined}
        >
          <span>
            first
          </span>
          <svg
            aria-label="link-down"
            className="grommetux-control-icon grommetux-control-icon-link-down grommetux-control-icon--responsive"
            height="24px"
            role="img"
            version="1.1"
            viewBox="0 0 24 24"
            width="24px"
          >
            <path
              d="M12,2 L12,22 M3,11 L12,2 L21,11"
              fill="none"
              stroke="#000"
              strokeWidth="2"
              transform="matrix(1 0 0 -1 0 24)"
            />
          </svg>
        </div>
      </button>
    </th>
    <th>
      <button
        aria-label={undefined}
        className="grommetux-button grommetux-button--fill grommetux-button--plain"
        disabled={false}
        href={undefined}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        type="button"
      >
        <div
          className="grommetux-box grommetux-box--direction-row grommetux-box--justify-start grommetux-box--align-center grommetux-box--responsive grommetux-box--pad-between-small"
          id={undefined}
          onClick={undefined}
          role={undefined}
          style={Object {}}
          tabIndex={undefined}
        >
          <span>
            second
          </span>
        </div>
      </button>
    </th>
  </tr>
</thead>
`;
